src/components/PoolFinder/index.tsx

```diff
@@ -15,71 +14,40 @@ import { AutoColumn, ColumnCenter } from '../Column'
 import { ButtonPrimary, ButtonDropwdown, ButtonDropwdownLight } from '../Button'
 
 import { useToken } from '../../contexts/Tokens'
-import { usePopups } from '../../contexts/Application'
-import { usePrevious } from '../../hooks'
 import { useWeb3React } from '@web3-react/core'
 import { useAddressBalance } from '../../contexts/Balances'
-import { usePair, useAllPairs } from '../../contexts/Pairs'
+import { useLocalStoragePairAdder } from '../../contexts/LocalStorage'
+import { useReserves } from '../../data/Reserves'
 
-function PoolFinder({ history }: RouteComponentProps) {
-  const Fields = {
-    TOKEN0: 0,
-    TOKEN1: 1
-  }
+const Fields = {
+  TOKEN0: 0,
+  TOKEN1: 1
+}
 
+function PoolFinder({ history }: RouteComponentProps) {
   const { account } = useWeb3React()
   const [showSearch, setShowSearch] = useState<boolean>(false)
   const [activeField, setActiveField] = useState<number>(Fields.TOKEN0)
 
-  const [, addPopup] = usePopups()
-
   const [token0Address, setToken0Address] = useState<string>()
   const [token1Address, setToken1Address] = useState<string>()
-
   const token0: Token = useToken(token0Address)
   const token1: Token = useToken(token1Address)
 
-  const pair: Pair = usePair(token0, token1)
-  const position: TokenAmount = useAddressBalance(account, pair?.liquidityToken)
-
-  const newPair: boolean = pair && JSBI.equal(pair.reserve0.raw, JSBI.BigInt(0))
-  const allowImport: boolean = position && JSBI.greaterThan(position.raw, JSBI.BigInt(0))
-
-  const allPairs = useAllPairs()
-  const pairCount = Object.keys(allPairs)?.length
-  const pairCountPrevious = usePrevious(pairCount)
-  const [newLiquidity, setNewLiquidity] = useState<boolean>(false) // check for unimported pair
+  const pair: Pair = useReserves(token0, token1)
```

Comment:
`useReserves` is an odd name for a function that returns a pair

Reply:
agreed, relic of old code, changed